### ISSUE TITLE
WebXR mode in React causing page error in Android Chrome

### ISSUE THREAD

**Issue by milesgreen** [google/model-viewer#1694]
### Description

I'm currently trying to debug the following issue. I'm seeing a browser page crash when entering WebXR mode on Android Chrome when using ModelViewer within React. The initial 3D model (non-ar) view is presented as expected. The page crash is observed when user taps AR button causing error before WebXR session starts.  

Page crash observed when `ar-modes` is set to `webxr scene-viewer quick-look`.

Page crash does not happen in same React web app in same browsers when `ar-modes` is set to `scene-viewer quick-look webxr`. SceneViewer starts as expected. 

When viewing WebXR ModelViewer tests outside of the React web app, using  `ar-modes` set to `webxr scene-viewer quick-look`, the same browsers work fine. WebXR session is seen working as expected. No page crash. 

Current hypothesis is that React is causing issues with WebXR.

Wondering if anyone else has demos of ModelViewer + WebXR working in React?

#### Live Demo
Hard to share a link for this one as it is behind access controls. Will see if I can make a basic public demo. 

### Version

- model-viewer: 1.1 and 1.3

### Browser Affected

- [x] Chrome (86)
- [ ] Edge
- [ ] Firefox
- [ ] IE
- [ ] Safari

### OS

- [x] Android (10 and 11)
- [ ] iOS
- [ ] Linux
- [ ] MacOS
- [ ] Windows

![image](https://user-images.githubusercontent.com/53870482/98237756-4f6e9980-1f5d-11eb-89bf-cb0a2810ed59.png)

**Comment by elalish** [comment]
Thanks, a basic public demo would be much appreciated! @klausw this looks like a chrome bug; think we can find a crash report?

**Comment by milesgreen** [comment]
As an update to this issue, I can share the crash report ID:

![](https://files.slack.com/files-pri/TKVHUUMQF-F01EMRGDXL6/chrome_crash-log.jpeg)

**Comment by milesgreen** [comment]
Also, prior to the page crash, I momentarily see the exit fullscreen instructions, only for a split second, and then the I get the crash screen. 

![Uploading screen-20201116-093137_exported_74249_1605540955920.jpg…]()

**Comment by klausw** [comment]
I haven't been able to find a crash report for ID fc74d5b44a68 in our system, but there may be a delay in processing for that. I'll check back in a bit to see if it shows up.

Could you please provide the Chrome browser version and OS version from chrome://version ?

Also, assuming you've been testing on the stable Chrome version, would you be willing to test on Chrome Dev or Canary to see if that also crashes?

**Comment by klausw** [comment]
@milesgreen , it looks like that crash report hasn't been uploaded yet. If I'm translating it right, it's showing a local ID. After uploading, it should say something like "Uploaded Crash Report ID" with a 16-hex-digit ID code instead of "Local crash context". I'd need that longer ID to look up the uploaded report. 

If you have a "provide additional details" button, could you use that to file a Chromium bug and link that here?

**Comment by milesgreen** [comment]
Thanks @klausw 

Original tests were done on Chrome 86.0.4240.198 / Android 10

Chrome Beta and Canary show same page crash. 

Will check now for the report to see if I can get the long ID.

**Comment by milesgreen** [comment]
Crash report has been submitted, here's the longer ID:

4101ad4eb0778b77

![WhatsApp Image 2020-11-16 at 19 34 53 (2)](https://user-images.githubusercontent.com/53870482/99317872-a417eb80-285e-11eb-9f48-40836fccbeef.jpeg)

**Comment by klausw** [comment]
Thank you, I got the crash report and backtrace, apparently this is happening when setting up fullscreen mode for the DOM overlay element. Are you doing anything unusual with the DOM overlay element or document when starting the session, for example deleting/recreating the element or its parents? It obviously shouldn't crash even if that does happen, but it could help in finding a workaround.

If you have a link to a site showing this behavior, it would be great to get that to confirm the cause and help verify a fix.

```
  | 0x00000000c228398c | (libmonochrome.so -scoped_refptr.h:231) |   | blink::GraphicsLayer::SetContentsOpaque(bool)
  | 0x00000000c2283a93 | (libmonochrome.so -composited_layer_mapping.cc:310) |   | blink::CompositedLayerMapping::UpdateContentsOpaque()
  | 0x00000000c23c8f9f | (libmonochrome.so -local_frame_view.cc:1852) |   | blink::LocalFrameView::SetBaseBackgroundColor(blink::Color const&)
  | 0x00000000c387bf9d | (libmonochrome.so -xr_system.cc:1430) |   | blink::XRSystem::OnRequestSessionReturned(blink::XRSystem::PendingRequestSessionQuery*, mojo::StructPtr<device::mojom::blink::RequestSessionResult>)
```

**Comment by klausw** [comment]
This should be fixed as of today's Chrome Canary 89.0.4329.2 or newer. Please let me know in case of regressions, I'll look into getting this merged to Chrome 88 also.

**Comment by milesgreen** [comment]
Testing...

**Comment by milesgreen** [comment]
Confirming that this is fixed on Android 10 and 11 in Chrome Canary 89.0.4329.2

Thanks

**Comment by danbaechtold** [comment]
We have this issue too, in a Vue.js App. Latest Chrome 88 on a Samsung tablet just crashes on model load (even before starting AR mode). This affects lots of our customers. Any news if (or when) this has been merged into Chrome 88? Thanks

**Comment by elalish** [comment]
@danbaechtold If yours is crashing on model load before starting AR, then it sounds unrelated to this WebXR issue. Can you open a new issue and fill out all the details regarding the problem you're seeing?
